store/tikv/prewrite.go

```diff
@@ -140,6 +151,23 @@ func (action actionPrewrite) handleSingleBatch(c *twoPhaseCommitter, bo *Backoff
 					c.run(c, nil)
 				}
 			}
+
+			if c.isOnePC() {
+				if prewriteResp.OnePcCommitTs == 0 {
+					logutil.Logger(bo.ctx).Warn("1pc failed and fallbacks to normal commit procedure",
+						zap.Uint64("satrTS", c.startTS))
+					tikvOnePCTxnCounterFallback.Inc()
+					c.setOnePC(false)
```

Comment:
TODO? I think there may be a `continue`?

Reply:
I means  to support falling back on TiKV side, which means, TiKV successfully prewrite the lock so (`len(keyErrs) == 0`) but without committing it with 1PC, and TiDB can know it here and continue committing it with 2PC protocol here. However currently TiKV will return CommitTsTooLarge error here, so this case should never happen, currently, I think.

Reply:
I mean, I didn't mean to `continue` here, but continue committing the transaction with 2pc protocol.

Reply:
Sorry, now I understand your idea. It makes sense.